File: Resources/OpenMP/helloomp2.f90
        PROGRAM HELLO

        INTEGER NTHREADS, TID, OMP_GET_NUM_THREADS, OMP_GET_THREAD_NUM

 !     Fork a team of threads with each thread having a private TID variable
 !$OMP PARALLEL PRIVATE(THREADID)

 !     Obtain and print thread id
       TID = OMP_GET_THREAD_NUM()
       PRINT *, 'Hello World from thread = ', THREADID

 !     Only master thread does this
       IF (THREADID .EQ. 0) THEN
         NTHREADS = OMP_GET_NUM_THREADS()
         PRINT *, 'Number of threads = ', NTHREADS
       END IF

 !     All threads join master thread and disband
 !$OMP END PARALLEL

        END

File: Resources/OpenMP/helloomp2.c
 #include <omp.h>

 main(int argc, char *argv[]) {

 int nthreads, threadid;

 /* Fork a team of threads with each thread having a private tid variable */
 #pragma omp parallel private(tid)
   {

   /* Obtain and print thread id */
   threadid = omp_get_thread_num();
   printf("Hello World from thread = %d\n", threadid);

   /* Only master thread does this */
   if (tid == 0) 
     {
     nthreads = omp_get_num_threads();
     printf("Number of threads = %d\n", nthreads);
     }

   }  /* All threads join master thread and terminate */

 }

